### ISSUE TITLE
Running in Node.js application

### ISSUE THREAD

**Issue by dakshshah96** [resin-io/resin-wifi-connect#210]
I love working with Resin.io but can't figure out how to integrate `resin-wifi-connect` in my existing Node.js application.

Currently, my Dockerfile looks like this:

```
FROM resin/raspberrypi-alpine-node:slim

WORKDIR /usr/src/app

COPY package.json package.json

RUN npm install

COPY . .

CMD ["node", "index.js"]
```

It'd be great if someone can help me figure out how to run `resin-wifi-connect` before my Node.js application starts. cc @majorz 

Would something like this work?

**Dockerfile**

```
FROM resin/raspberrypi-alpine-node:slim

ENV INITSYSTEM on

RUN apt-get update \
  && apt-get install -y dnsmasq wireless-tools \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/*

WORKDIR /usr/src/app

RUN curl https://api.github.com/repos/resin-io/resin-wifi-connect/releases/latest -s \
  | grep -hoP 'browser_download_url": "\K.*%%RESIN_ARCH%%\.tar\.gz' \
  | xargs -n1 curl -Ls \
  | tar -xvz -C /usr/src/app/

COPY scripts/start.sh .

CMD ["bash", "start.sh"]
```

**start.sh**

```
#!/usr/bin/env bash

export DBUS_SYSTEM_BUS_ADDRESS=unix:path=/host/run/dbus/system_bus_socket

# Choose a condition for running WiFi Connect according to your use case:

# 1. Is there a default gateway?
# ip route | grep default

# 2. Is there Internet connectivity?
# nmcli -t g | grep full

# 3. Is there Internet connectivity via a google ping?
# wget --spider http://google.com 2>&1

# 4. Is there an active WiFi connection?
iwgetid -r

if [ $? -eq 0 ]; then
    printf 'Skipping WiFi Connect\n'
else
    printf 'Starting WiFi Connect\n'
    ./wifi-connect
fi

# Start your application here.
npm install
node main.js
```

I don't know how to start my Node.js from the script file.

**Comment by majorz** [comment]
Alpine relies on musl libc and will not work with WiFi Connect. Also it does not have apt-get. You can use a Docker image based on Debian instead. We provide base images which target Node.js development. What type of device specifically you are using - is it a raspberry pi 3?

**Comment by dakshshah96** [comment]
@majorz Yes, I'm using a Raspberry Pi 3. Is the `resin/raspberrypi3-node` base image fine?

**Comment by majorz** [comment]
I am away now and could not try running that, but you may go to the releases page, get the link to the armv7hf archive and do ´curl -OL https://...tar.gz´ to download and then ´tar zxf ...tar.gz´ to extract it. I can check it out tomorrow if you cannot resolve it.

**Comment by dakshshah96** [comment]
@majorz I did that on my Terminal. `curl` downloads the tar.gz file properly and I was able to extract the contents as well with `tar zxf ...` without any errors.

**Comment by jgentes** [comment]
@dakshshah96 we run our node app in a separate container and it works great. There's no need to map any network connectivity between the containers.

**Comment by majorz** [comment]
@dakshshah96 I will close this now, but please reopen it or open a new issue in case you encounter difficulties or problems.

**Comment by dakshshah96** [comment]
@majorz Still facing the same problem 😞

**Comment by majorz** [comment]
I will try to find time to look at it tomorrow.

**Comment by SeJohnEff** [comment]
I tried as @jgentes proposed . Using one container for wifi connect (on debian) & on the other container simple-server-node. 
WiFi connect and Simple server works fine in known WiFi network (Can see "Skipping WiFi Connect" in log) . 
Does not work on unknown WiFi network . It is hard to troubleshoot but I guess I need to give permission to run as WiFi hotspot?

---------- dockercompose.yml ---------
version: '2'
services:
  server:
    build: ./simple-server-node
    ports:
      - "80"

  wificonnect:
    build: ./resin-wifi-connect
# mmust run in priviliged mode to be able to change hostname
    privileged: true
    network_mode: host

**Comment by wenzou** [comment]
same problem here
